### PR TITLE
chore(Storybook): update input fields stories, unit, acceptance and screenshot tests

### PR DESCRIPTION
**Pull request by marcoskolodny**
Continuation of the storybook refactor: [Issue](https://jira.tid.es/browse/WEB-819)

### CODE REVIEW

File: src/text-field-base.css.ts [hunk 1]
```diff
@@ -72,16 +72,16 @@ export const textArea = style([
 ]);
 
 export const textAreaWithLabel = style({
-    marginTop: 28,
+    paddingTop: 28,
```
Comment (marcoskolodny) [inline src/text-field-base.css.ts:75]: Unrelated change. This was causing the upper area of a TextField to not be part of the text area if multiline is set to true

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-text-field-multiline-2-snap.png
Comment (yceballost) [inline src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-text-field-multiline-2-snap.png:1]: could we include and example with long text to see if the text cover the width?

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-text-field-read-only-1-snap.png
Comment (yceballost) [inline src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-text-field-read-only-1-snap.png:1]: readOnly + focus state is missing is screenshot

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-cvv-field-1-snap.png
Comment (yceballost) [inline src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-cvv-field-1-snap.png:1]: missing a screenshot of cvv tooltip

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-password-field-2-snap.png
Comment (yceballost) [inline src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-password-field-2-snap.png:1]: this screenshot is not using Lucida Grande as a font

File: src/__acceptance_tests__/input-fields-acceptance-test.tsx
Comment (yceballost) [inline src/__acceptance_tests__/input-fields-acceptance-test.tsx:1]: Should we have desktop screenshots? 🤔

Comment (yceballost) [inline src/__acceptance_tests__/input-fields-acceptance-test.tsx:1]: I know select components is a different component but there are very similarities between input fields, should we include select in this pr? just asking

**Review by yceballost — CHANGES_REQUESTED** [review]

File: src/__acceptance_tests__/input-fields-acceptance-test.tsx
Comment (marcoskolodny) [reply src/__acceptance_tests__/input-fields-acceptance-test.tsx:1]: It's one of the pending components, I will do it in a different iteration to avoid increasing the amount of changes per PR. You can see the updated list of ready/pending/reviewed components in the issue's description: [Link](https://jira.tid.es/browse/WEB-819)

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-text-field-multiline-2-snap.png
Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-text-field-multiline-2-snap.png:1]: Added

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-text-field-read-only-1-snap.png
Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-text-field-read-only-1-snap.png:1]: Added

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-cvv-field-1-snap.png
Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-cvv-field-1-snap.png:1]: Added. Notice that it's using the old version of the tooltip, so it will be "broken" visually until we merge the tooltip PR

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-password-field-2-snap.png
Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-password-field-2-snap.png:1]: And it won't, because that font is not installed in the docker where the tests are executed. But there was a hack for using the inherited font instead of the system one in acceptance tests. I removed it in order to keep consistency with other components that use system font for tests (like PinField).

File: src/__acceptance_tests__/input-fields-acceptance-test.tsx
Comment (marcoskolodny) [reply src/__acceptance_tests__/input-fields-acceptance-test.tsx:1]: We don't see a real value in including desktop screenshots for ALL the fields, given that the changes between mobile and desktop are only in the fields' size, and also all the fields use the same logic behind.

I added one TextField screenshot in desktop.

**Review by yceballost — APPROVED** [review]

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-cvv-field-icon-tooltip-appears-properly-1-snap.png
[the remark this answers is not in the document]
Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-cvv-field-icon-tooltip-appears-properly-1-snap.png:1]: This is actually "expected". Remember there is a PR with the new version of the tooltip. Given that this branch doesn't have those updates, it will be "broken" for now

File: src/__stories__/credit-card-expiration-field-story.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                    </Stack>
+                </Stack>
+            </Box>
+        </ResponsiveLayout>
+    );
+};
+
+Uncontrolled.storyName = 'uncontrolled';
+Uncontrolled.args = {
+    defaultValue: '13/21',
```
[the remark this answers is not in the document]
Comment (marcoskolodny) [reply src/__stories__/credit-card-expiration-field-story.tsx:130]: well, if you want the page to load with a particular value already written in the field, how would you do it otherwise? For example, it's comfortable for screenshot/acceptance tests.

File: src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-cvv-field-icon-tooltip-appears-properly-1-snap.png
[the remark this answers is not in the document]
Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/input-fields-screenshot-test-tsx-cvv-field-icon-tooltip-appears-properly-1-snap.png:1]: Anyways, I will remove this screenshot test from this PR because, after discussing with @aweell, this component's behavior regarding its tooltip will change, so it makes more sense to add this test when we update the component with the new specs.

File: src/__acceptance_tests__/input-fields-acceptance-test.tsx [hunk 3]
```diff
@@ -193,9 +180,9 @@ test.each(STORY_TYPES)('PhoneNumberField (%s)', async (storyType) => {
 });
 
 test.each(STORY_TYPES)('SearchField (%s)', async (storyType) => {
-    const page = await openStoryPage(getStoryOfType(storyType));
+    const page = await openStoryPage(getStoryOfType('searchfield', storyType));
 
-    const field = await screen.findByLabelText('Search');
+    const field = await screen.findByTestId('search-field');
```
[the remark this answers is not in the document]
Comment (marcoskolodny) [reply src/__acceptance_tests__/input-fields-acceptance-test.tsx:185]: good point, I changed it to use findByLabelText in all the acceptance tests

File: src/__stories__/credit-card-expiration-field-story.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    return (
+        <ResponsiveLayout isInverse={inverse} fullWidth>
+            <Box padding={16}>
+                <Stack space={16}>
+                    <div data-testid="credit-card-expiration-field-wrapper">
+                        <CreditCardExpirationField
+                            value={rawValue}
+                            onChange={(e) => setRawValue(e.target.value)}
+                            onChangeValue={(value) => setValue(value)}
```
[the remark this answers is not in the document]
Comment (marcoskolodny) [reply src/__stories__/credit-card-expiration-field-story.tsx:50]: Good idea, I'll do that

**Review by atabel — APPROVED** [review]

**Review by pladaria — APPROVED** [review]
